### PR TITLE
[IMP] website: Show Scroll in case of full height

### PR DESCRIPTION
**Pull request by oussamamessaoudi**
- Show "scroll down button" only when height is 100%.
- Make the "scroll down button" a child of "height" option.

task-2471323

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/snippets/s_map.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
             <we-checkbox string="Description" data-no-preview="true" data-show-description="true"/>
         </div>
     </xpath>
-    <xpath expr="//div[@data-option-name='minHeight']" position="inside">
+    <xpath expr="//div[@data-js='ScrollButton']//we-button-group" position="after">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_map.xml:56]: After this change, you add an option inside an option... if it does not crashes, it is pure luck

File: addons/website/static/src/js/editor/snippets.options.js [hunk 2]
```diff
@@ -2681,6 +2681,21 @@ options.registry.ScrollButton = options.Class.extend({
...
+         await this._super(...arguments);
+         if (params.isFullHeight) {
+             this.$button.show();
+         } else {
+             if (previewMode) {
+                 this.$button.hide();
+             } else {
+                 this.$button.detach();
+             }
+         }
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:2697]: This should only apply to the specific height class control, not all potential selec-class that could be added in that option. You can handle that by using two methods on the same we-button (so here data-select-class for the height + data-... to handle the button hide/detach).

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/views/snippets/s_map.xml (hunk 1 above)
Comment (qsm-odoo) [reply addons/website/views/snippets/s_map.xml:56]: Why are you adding a smiley which laughs at my comment and keep the PR unchanged..? What am I missing? :grimacing: 

It is actually already an error in the original code but fix it for master please :grimacing:

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/js/editor/snippets.options.js [hunk 3]
```diff
@@ -2747,6 +2747,21 @@ options.registry.ScrollButton = options.Class.extend({
     // Options
     //--------------------------------------------------------------------------
 
+    /**
+     *
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:2751]: Remove that extra line

File: addons/website/views/snippets/snippets.xml [hunk 4]
```diff
@@ -1001,20 +1001,19 @@
         </we-button-group>
     </div>
 
-    <!-- Min height of section -->
-    <div data-option-name="minHeight" data-selector="section">
```
Comment (qsm-odoo) [inline addons/website/views/snippets/snippets.xml:1005]: You removed that data-option-name without adapting the code related to it: I grepped it, there are some occurences in themes, so you need to adapt that in PR with the same name in those theme repo (or found another solution here?). To check.

File: addons/website/views/snippets/snippets.xml [hunk 5]
```diff
@@ -1001,20 +1001,19 @@
...
-        <we-button-group string="Height">
+    <!-- Scroll to next section button (only for full height) -->
+    <div data-js="ScrollButton" data-selector="section">
+        <!-- Min height of section -->
+        <we-button-group string="Height" data-show-scroll-button="">
             <we-button data-name="minheight_auto_opt" data-select-class="" title="Fit content">Auto</we-button>
             <we-button data-select-class="o_half_screen_height" title="Half screen">50%</we-button>
-            <we-button data-select-class="o_full_screen_height" title="Full screen">100%</we-button>
+            <we-button data-select-class="o_full_screen_height" title="Full screen"  data-name="full_height_opt" data-show-scroll-button="true">100%</we-button>
```
Comment (qsm-odoo) [inline addons/website/views/snippets/snippets.xml:1010]: There is a double whitespace

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
Was about to merge but there are some problems, can you check? Thanks :+1: 

Also, commit message: no uppercase to start the title please

File: addons/website/views/snippets/snippets.xml (hunk 4 above)
Comment (oussamamessaoudi) [reply addons/website/views/snippets/snippets.xml:1005]: I checked but couldn't find any occurrence of `data-option-name` or `minHeight` in `design-themes` repo.

Comment (qsm-odoo) [reply addons/website/views/snippets/snippets.xml:1006]: How did you check? :roll_eyes: 

Screenshot from 2021-06-14 11-47-51

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by qsm-odoo** [comment]
@robodoo r+
